src/lib/utils.py

```diff
@@ -68,6 +68,14 @@ def _extract_timestamp(info):
     raise CheckerRemoteError(f"Cannot parse date/time: {date_str}")
 
 
+def _check_newline(manifest_path):
+    with open(manifest_path, "r") as f:
+        if f.read()[-1] == "\n":
```

Comment:
This could be done in a less expensive way by seeking the file to the end and reading only the last char.